libraries/AP_OSD/AP_OSD.h

```diff
@@ -190,10 +236,140 @@ class AP_OSD_Screen
     void draw_clk(uint8_t x, uint8_t y);
 };
 
+/*
+  class to hold one setting
+ */
+class AP_OSD_ParamSetting : public AP_OSD_Setting
+{
+public:
```

Comment:
Please check visibility on members on this class.  Pretty sure some of them can be private.

Reply:
I already tried that, pretty sure they all have to be public unless we make this class a friend of AP_OSD_ParamScreen, which seems like its defeating the point.

Reply:
I tell a lie - _metadata_index can be private, but that is all